src/integrationTest/java/com/mongodb/hibernate/type/temporal/UnsupportedItems.java

```diff
@@ -0,0 +1,254 @@
+    }
+
+    @Entity
+    static class ItemWithNestedAggregateEmbeddableWithCollectionPersistentAttribute<T> {
+        @Id
+        int id;
+
+        AggregateEmbeddableNestedWithCollectionPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithNestedCollectionOfAggregateEmbeddable<T> {
+        @Id
+        int id;
+
+        AggregateEmbeddableWithCollectionOfAggregateEmbeddableWithBasicPersistentAttribute<T> v;
+    }
+
+    @Embeddable
+    static class FlattenedEmbeddableWithBasicPersistentAttribute<T> {
+        T v;
+    }
+
+    @Embeddable
+    private static class FlattenedEmbeddableWithArrayPersistentAttribute<T> {
+        T[] v;
+    }
+
+    @Embeddable
+    static class FlattenedEmbeddableWithCollectionPersistentAttribute<T> {
+        Collection<T> v;
+    }
+
+    @Embeddable
+    static class FlattenedEmbeddableNestedWithBasicPersistentAttribute<T> {
+        FlattenedEmbeddableWithBasicPersistentAttribute<T> v;
+    }
+
+    @Embeddable
+    static class FlattenedEmbeddableNestedWithArrayPersistentAttribute<T> {
+        FlattenedEmbeddableWithArrayPersistentAttribute<T> v;
+    }
+
+    @Embeddable
+    static class FlattenedEmbeddableNestedWithCollectionPersistentAttribute<T> {
+        FlattenedEmbeddableWithCollectionPersistentAttribute<T> v;
+    }
+
+    @Struct(name = "AggregateEmbeddableWithBasicPersistentAttribute")
+    @Embeddable
+    static class AggregateEmbeddableWithBasicPersistentAttribute<T> {
+        T v;
+    }
+
+    @Struct(name = "AggregateEmbeddableWithArrayPersistentAttribute")
+    @Embeddable
+    private static class AggregateEmbeddableWithArrayPersistentAttribute<T> {
+        T[] v;
+    }
+
+    @Struct(name = "AggregateEmbeddableWithCollectionPersistentAttribute")
+    @Embeddable
+    static class AggregateEmbeddableWithCollectionPersistentAttribute<T> {
+        Collection<T> v;
+    }
+
+    @Struct(name = "AggregateEmbeddableNestedWithBasicPersistentAttribute")
+    @Embeddable
+    static class AggregateEmbeddableNestedWithBasicPersistentAttribute<T> {
+        AggregateEmbeddableWithBasicPersistentAttribute<T> v;
+    }
+
+    @Struct(name = "AggregateEmbeddableNestedWithArrayPersistentAttribute")
+    @Embeddable
+    static class AggregateEmbeddableNestedWithArrayPersistentAttribute<T> {
+        AggregateEmbeddableWithArrayPersistentAttribute<T> v;
+    }
+
+    @Struct(name = "AggregateEmbeddableNestedWithCollectionPersistentAttribute")
+    @Embeddable
+    static class AggregateEmbeddableNestedWithCollectionPersistentAttribute<T> {
+        AggregateEmbeddableWithCollectionPersistentAttribute<T> v;
+    }
+
+    @Struct(name = "AggregateEmbeddableWithCollectionOfAggregateEmbeddableWithBasicPersistentAttribute")
+    @Embeddable
+    static class AggregateEmbeddableWithCollectionOfAggregateEmbeddableWithBasicPersistentAttribute<T> {
+        Collection<AggregateEmbeddableWithBasicPersistentAttribute<T>> v;
+    }
```

Comment:
The name of this struct should have stayed what it was: `AggregateEmbeddableWithCollectionOfAggregateEmbeddablePersistentAttribute`. Here is that same name with parenthesis and spaces that help see the meaning: `AggregateEmbeddable With (Collection Of AggregateEmbeddable) PersistentAttribute`. That is, it's an aggregate embeddable with a persistent attribute of the `(Collection Of AggregateEmbeddable)` type.

The new name `AggregateEmbeddableWithCollectionOfAggregateEmbeddableWithBasicPersistentAttribute` is `AggregateEmbeddable With (Collection Of (AggregateEmbeddable With Basic)) PersistentAttribute`, which clearly is illformed, as `(AggregateEmbeddable With Basic)` is illformed. Also, the type of persistent attributes of the aggregate embeddable elements in the collection is irrelevant.